libcalico-go/lib/backend/k8s/resources/kubecontrollersconfig.go

```diff
@@ -17,18 +17,23 @@ package resources
 import (
 	"reflect"
 
-	apiv3 "github.com/projectcalico/api/pkg/apis/projectcalico/v3"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/go-playground/validator.v9"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
+
+	apiv3 "github.com/projectcalico/api/pkg/apis/projectcalico/v3"
 )
 
 const (
 	KubeControllersConfigResourceName = "KubeControllersConfigurations"
 	KubeControllersConfigCRDName      = "kubecontrollersconfigurations.crd.projectcalico.org"
 )
 
+var validate *validator.Validate
```

Comment:
Though I like the pattern of adding a validator object in order to make it extensible to other resources defined by the `customResourceClient`, I think that the validator we use here can be much simpler than the validator we use for the client in general. I think it might be easier (and a little cleaner) to define a separate validator here that will just have a `Validate` method that is overridden per resource. So in our `KubeControllersConfig` case, `Validate` would only need to validate the `k8sSyncLabels` field and we wouldn't need to use an `init` call or a separate call to `Var` in order to target the specific fields to validate.